app/components/data-model/DataModel.js

```diff
@@ -32,14 +32,39 @@
     
     module.service('DataModelService', function() {
```

Comment:
It is better to split the service into 2 services: users and messages.
In this case it will be easier to test them